cmd/nerdctl/container_run_soci_linux_test.go

```diff
@@ -0,0 +1,91 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/containerd/nerdctl/pkg/testutil"
+)
+
+func TestRunSoci(t *testing.T) {
+	//TODO: validate that mounts match the mounts of soci rpull
+	testutil.DockerIncompatible(t)
+	base := testutil.NewBase(t)
+	requiresSoci(base)
+
+	//finding soci executable
+	sociExecutable, err := exec.LookPath("soci")
+	base.T.Logf("soci executable path: %s", sociExecutable) //DEBUG
+	if err != nil {
+		t.Fatalf("SOCI is not installed.")
+	}
+
+	//running rpull
+	rpullCmd := exec.Command(sociExecutable, []string{"image", "rpull", testutil.FfmpegSociImage}...)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	rpullCmd.Env = os.Environ()
+
+	base.T.Logf("running rpull") //DEBUG
+	err = rpullCmd.Run()
+	if err != nil {
+		t.Fatalf(err.Error()) //DEBUG
+	}
+
```

Comment:
Maybe use `rmi` to clean up the `rpull` before `nerdctl pull`?

Reply:
Cleaning up the image shouldn't be an issue since soci's rpull uses a different storage space than nerdctl. I'm also not sure how to remove the image after it's pulled by SOCI since the SOCI cli doesn't include a command for removing images as far as I know. Just to be sure though, I switched the order of the commands so the I run `nerdctl run...` first before `rpull`, and I stopped the container after the counting the mounts for `nerdctl run...` so that the image is removed before doing `rpull`.

Reply:
Stopping container doesn't clean up the image? Don't we need to do `rmi`?

Reply:
You're right. I mistook the `--rm` for removing the image and container after the container is stopped. I've added `rmi` to the test to remove the image.